files/lib/data/faq/QuestionAction.class.php

```diff
@@ -13,6 +13,10 @@
 use wcf\system\search\SearchIndexManager;
 use wcf\system\WCF;
 
+/**
+ * @method  QuestionEditor[] getObjects()
+ * @method  QuestionEditor   getSingleObject()
```

Comment:
Durch die Vererbung ist die Angabe nicht notwendig

Reply:
Jein: Natürlich sind diese Funktionen in der Klasse `AbstractDatabaseObjectAction` enthalten, aber mit dem Rückgabetyp `DatabaseObjectEditor[]` / `DatabaseObjectEditor`. Die IDE kann nicht wissen, dass diese zwei Funktionen im Falle von `QuestionAction` den Rückgabetyp `QuestionEditor[]` / `QuestionEditor` haben, sodass Auto Completion für Funktionen, die in `QuestionEditor` existieren, aber nicht in `DatabaseObjectEditor` funktionieren, nicht möglich wäre und die IDE einen Fehler anzeigen würde/sollte, da man auf eine nicht existierende Funktion zugreifen würde. Prinzipiell ist es auch Best Practice von WoltLab, diese Typisierung immer in Form von PHPDoc anzugeben.